File: src/main/java/me/chriss99/CutOutRectangleTLM.java
package me.chriss99;

import org.joml.Vector2f;
import org.joml.Vector2i;

import java.util.Collection;

public class CutOutRectangleTLM<T> extends SquareTLM<T> {
    protected Area skipArea;

    public CutOutRectangleTLM(int renderDistance, Vector2f initPos, Area skipArea) {
        super(renderDistance, initPos);
        this.skipArea = skipArea.copy();
    }

    @Override
    public boolean loadPolicy(Vector2i tilePos, T tile) {
        return super.loadPolicy(tilePos, tile) && !skipArea.contains(tilePos);
    }

    @Override
    public Collection<Vector2i> loadCommander() {
        Collection<Vector2i> toLoad = super.loadCommander();
        toLoad.removeIf(v -> skipArea.contains(v));
        return toLoad;
    }

    public void setSkipArea(Area skipArea) {
        this.skipArea = skipArea;
    }
}

File: src/main/kotlin/CutOutRectangleTLM.kt
package me.chriss99

import glm_.vec2.Vec2
import glm_.vec2.Vec2i

class CutOutRectangleTLM<T>(renderDistance: Int, initPos: Vec2, var skipArea: Area) : SquareTLM<T>(renderDistance, initPos) {
    override fun loadPolicy(tilePos: Vec2i, tile: T) = super.loadPolicy(tilePos, tile) && tilePos !in skipArea
    override fun loadCommander() = super.loadCommander().filter { it !in skipArea }
}